examples/inclined_plane.py

```diff
@@ -0,0 +1,130 @@
+data, she wants to infer the coefficient of friction mu between the box and the inclined
+plane. She already has (deterministic) python code that can simulate the amount of time
+that it takes the little box to slide down the inclined plane as a function of mu. Using
+Pyro, she can reverse the simulator and infer mu from the observed descent times.
+"""
+
+little_g = 9.8  # m/s/s
+mu0 = 0.12  # actual coefficient of friction in the experiment
+time_measurement_sigma = 0.02  # observation noise in seconds (known quantity)
+
+
+# the forward simulator, which does numerical integration of the equations of motion
+# in steps of size dt, and optionally includes measurement noise
+
+def simulate(mu, length=2.0, phi=np.pi / 6.0, dt=0.005, noise_sigma=None):
+    T = Variable(torch.zeros(1))
+    velocity = Variable(torch.zeros(1))
+    displacement = Variable(torch.zeros(1))
+    acceleration = Variable(torch.Tensor([little_g * np.sin(phi)])) - \
+        Variable(torch.Tensor([little_g * np.cos(phi)])) * mu
+
+    if acceleration.data[0] <= 0.0:             # the box doesn't slide if the friction is too large
+        return Variable(torch.Tensor([1.0e5]))  # return a very large time instead of infinity
+
+    while displacement.data[0] < length:  # otherwise slide to the end of the inclined plane
+        displacement += velocity * dt
+        velocity += acceleration * dt
+        T += dt
+
+    if noise_sigma is None:
+        return T
+    else:
+        return T + Variable(noise_sigma * torch.randn(1))
+
+
+# analytic formula that the simulator above is computing via
+# numerical integration (no measurement noise)
+
+def analytic_T(mu, length=2.0, phi=np.pi / 6.0):
+    numerator = 2.0 * length
+    denominator = little_g * (np.sin(phi) - mu * np.cos(phi))
+    return np.sqrt(numerator / denominator)
+
+
+# generate N_obs observations using simulator and the true coefficient of friction mu0
+print("generating simulated data using the true coefficient of friction %.3f" % mu0)
+N_obs = 20
+torch.manual_seed(2)
+observed_data = torch.cat([simulate(Variable(torch.Tensor([mu0])), noise_sigma=time_measurement_sigma)
+                           for _ in range(N_obs)])
+observed_mean = np.mean([T.data[0] for T in observed_data])
+
+
+# define model with uniform prior on mu and gaussian noise on the descent time
+def model(observed_data):
+    mu_prior = Uniform(Variable(torch.zeros(1)), Variable(torch.ones(1)))
+    mu = pyro.sample("mu", mu_prior)
+
+    def observe_T(T_obs, obs_name):
+        T_simulated = simulate(mu)
+        T_obs_dist = Normal(T_simulated, Variable(torch.Tensor([time_measurement_sigma])))
+        pyro.observe(obs_name, T_obs_dist, T_obs)
+
+    for i, T_obs in enumerate(observed_data):
+        observe_T(T_obs, "obs_%d" % i)
+
+    return mu
+
+
+def main():
+    parser = argparse.ArgumentParser(description="parse args")
+    parser.add_argument('-n', '--num-samples', default=500, type=int)
```

Comment:
Could you please add a case in `tests/test_examples.py` so that this example will run during testing:
```diff
  def discover_examples():
      for root, dirs, files in os.walk(EXAMPLES_DIR):
          for basename in files:
              ...
              if '--num-epochs' in text:
                  args = ['--num-epochs=1']
              elif '--num-steps' in text:
                  args = ['--num-steps=1']
+             elif '--num-samples' in text:
+                 args = ['--num-samples=1']
              else:
                  # Either this is not a main file, or we don't know how to run it cheaply.
                  continue
              ...
```

Reply:
thanks, done